### DISCUSSION TITLE
How can i get Google Api from this library?

### DISCUSSION THREAD

**Discussion by hrynevychroman** [NathanAP/vue-google-maps-community-fork#82] [Q&A]
Hello everyone, I need to check if my marker inside polygon, and I'm using this function
```js
export function ifPointInsidePolygon(
  place?: GeocoderResult,
  polygonOptions?: google.maps.PolygonOptions
) {
  if (isNil(place) || isNil(polygonOptions)) return false;

  return google.maps.geometry.poly.containsLocation(
    place.geometry.location,
    new google.maps.Polygon(polygonOptions)
  );
}
```

But this don't work, because google.maps are not defined, how can I get google API from this library?

**Comment by ccencisoj** [comment]
This working for me, call the function '$gmapApiPromiseLazy()'

```typescript
app.use(VueGoogleMaps, {
  load: {
    key: env.GOOGLE_API_KEY,
    libraries: 'places,geometry',
  },
});
app.$gmapApiPromiseLazy();
```

**Reply by Stripkip** [reply]
_Disclaimer: I'm new to Vue and Nuxt as well, so any help is greatly appreciated!_

In my case I want to do fitBounds, and I have a similar problem with this line:
`const bounds = new google.maps.LatLngBounds();`

I'm using it with Nuxt 3, and with that I cannot do the solution with `app.use` and `app.$gmapApiPromiseLazy()`, since I'm not using `app.use`. Maybe I have to set this up at a `defineNuxtPlugin`-function, where you can do `nuxtApp.vueApp.use` or something?

Overall, I do have this plugin working, with map and markers and custom functions upon hovering over and clicking on markers. This is the final hurdle to complete my implementation.

**Reply by Stripkip** [reply]
I've fixed this last week, so I'll post here what I did for prosperity.
My project is in Vue 3 with Nuxt 3 with typescript.

What I did was that I declared the `google`-object globally in my project.

In the the project root I've made this file: `google-maps.d.ts`
The content of that file was just this line:
```
declare const google: typeof import("@types/google.maps").google;
```

Now it recognized `google.maps` and it was not longer undefined.
Weird that other people do not have this issue. I think it has to do with Nuxt, but I'm not sure about that though...

**Reply by Stripkip** [reply]
To add: if you're using node and in case it cannot find `@types/google.maps` to reference to, you can add it to your project by executing this line in bash in your project root:
`npm install --save-dev @types/google.maps`
